src/record_ts.h

```diff
@@ -428,6 +431,8 @@ struct RecordThreadState : ThreadState {
     void reduce_expanded(VarType vt, ReduceOp reduce_op, void *data,
                          uint32_t exp, uint32_t size) override;
 
+    void notify_opaque_width(uint32_t index, uint32_t width_index) override;
```

Comment:
Maybe a brief line comment, e.g.:

``Some kernels use the width of an array in a computation. When using the kernel freezing feature, this requires special precautions to ensure that the resulting capture remains usable with different array sizes. This notification function exists so that this special-case handling can be realized.``

Likely this should go into the top level ThreadState header declaration.